app/controllers/InviteController.scala

```diff
@@ -16,16 +16,23 @@
 
 package controllers
 
-import com.google.inject.Singleton
+import com.google.inject.{Inject, Singleton}
+import connectors.SubscriptionConnector
+import controllers.actions.AuthAction
+import models.{PsaSubscriptionDetails, SubscriptionDetails}
 import play.api.mvc.{AnyContent, Action}
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
 @Singleton
-class InviteController extends FrontendController {
+class InviteController@Inject()(authenticate: AuthAction,connector:SubscriptionConnector) extends FrontendController {
 
-  def onPageLoad: Action[AnyContent] = Action{
+  def onPageLoad: Action[AnyContent] = authenticate.async{
     implicit request =>
-      Ok
+      connector.getSubscriptionDetails(request.psaId.id) map { subscriptionDetails =>
+        subscriptionDetails.psaSubscriptionDetails.isPSASuspension match {
```

Comment:
maybe better to use an `IF` statement rather than pattern matching this?

Reply:
Agree, just for boolean check,  if else is better than pattern matching. Updated the same.